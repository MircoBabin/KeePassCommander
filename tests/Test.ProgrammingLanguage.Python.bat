@echo off
    setlocal
    call "%~dp0Utils\Utils.test-setup.bat" "%~f0"

    chcp 65001 > nul 2>&1
    set PYTHONIOENCODING=utf-8
    set PYTHONPYCACHEPREFIX=%tmp
    set PYTHONDONTWRITEBYTECODE=1
    python "%~dp0Test.ProgrammingLanguage.Python.py" "%KeePassCommandExe%" > "%KeePassCommand_TestOutput%"
    
    call "%~dp0Utils\Utils.diff.bat"
    if errorlevel 1 exit /b 1
    exit /b 0