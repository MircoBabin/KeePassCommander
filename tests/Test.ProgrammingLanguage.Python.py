import inspect, os.path
import sys

# begin: python 3 deprecated imp.load_source

if sys.version_info >= (3, 0):
    from importlib.machinery import SourceFileLoader
else:
    import imp
# end: python 3 deprecated imp.load_source


import binascii
import json

if len(sys.argv) < 2:
    raise Exception(
        "Provide full path to KeePassCommand.exe as first commandline parameter"
    )
KeePassCommandExe = sys.argv[1]

# get path of this file Test.ProgrammingLanguage.Python.py

module_filename = inspect.getframeinfo(inspect.currentframe()).filename
module_path = os.path.dirname(os.path.abspath(module_filename))
KeePassEntry_py = os.path.join(
    module_path, "..\\src\ProgrammingLanguagesConnectors\\KeePassEntry.py"
)

# load KeePassEntry.py containing function KeePassEntry
# begin: python 3 deprecated imp.load_source

KeePassEntryModule = None
if sys.version_info >= (3, 0):
    KeePassEntryModule = SourceFileLoader(
        "KeePassEntryModule", KeePassEntry_py
    ).load_module()
else:
    KeePassEntryModule = imp.load_source("KeePassEntryModule", KeePassEntry_py)
# end: python 3 deprecated imp.load_source

# BEGIN example

options = dict()
options["KeePassCommandExe"] = KeePassCommandExe
options["FieldNames"] = ["extra field 1", "extra password 1"]
options["AttachmentNames"] = ["example_attachment.txt"]

entry = KeePassEntryModule.KeePassEntry("Sample Entry", options)


class JsonBytesEncoder(json.JSONEncoder):
    def default(self, obj):
        if isinstance(obj, bytes):
            return binascii.hexlify(obj, " ").decode("utf-8")
        return json.JSONEncoder.default(self, obj)


print(json.dumps(entry, sort_keys=True, separators=(",", ":"), cls=JsonBytesEncoder))
