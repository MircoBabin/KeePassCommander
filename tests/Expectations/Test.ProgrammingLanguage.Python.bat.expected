{"attachments":[{"name":"example_attachment.txt","value":"54 68 69 73 20 69 73 20 61 6e 20 61 74 74 61 63 68 6d 65 6e 74 2e 0d 0a 0d 0a 57 69 74 68 20 74 68 72 65 65 20 6c 69 6e 65 73 20 69 6e 20 69 74 2e"}],"fields":[{"name":"extra field 1","value":"Some extra data"},{"name":"extra password 1","value":"Some extra password"}],"notes":"Notes \u00c9\u00e9n \u20ac\u00fbr\u00f2","password":"Password","title":"Sample Entry","url":"https://keepass.info/","urlhost":"keepass.info","urlpath":"/","urlport":"443","urlscheme":"https","username":"User Name"}
